lesson2/homework/helloworld/src/main/java/Animal.java

```diff
@@ -95,16 +98,18 @@ public void grow() {
 
 public class Farm {
 
+    private ArrayList<Animal> animals;
+
     public Farm(){
-        ArrayList animals = new ArrayList();
+        animals = new ArrayList<>();
     }
-    public void addAnimal(Object n_animal){
+    public void addAnimal(Animal n_animal){
```

Comment:
Instead of a generic `Object`, you should make your function parameter a specific `Animal`.  Otherwise, you could pass literally any Java object into this function, which is not what we want!